src/main/java/com/mwt/explorers/GenericExplorer.java

```diff
@@ -68,14 +68,16 @@ public DecisionTuple chooseAction(long saltedSeed, T context) {
     float sum = 0.f;
     float actionProbability = 0.f;
     int actionIndex = numWeights - 1;
-    for (int i = 0; i < numWeights; i++) {
-      float prob = weights.get(i) / total;
-      sum += prob;
+    int i = 0;
+    for (float weight: weights) {
```

Comment:
Act as if we only have linear access time because we don't need random access to List.